src/data_lakehouse_ingest/config_loader.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+from typing import Optional, Dict, Any, List, Union
+import json
+import logging
+import os
+from minio.error import S3Error
+
+
+class ConfigLoader:
+    """
+    ConfigLoader for minimal data-lakehouse ingestion configuration.
+
+    Supports loading from:
+      - Local file path
+      - Inline JSON string
+      - MinIO (s3a://bucket/key)
+    """
+
+    def __init__(
+        self,
+        cfg: Union[str, Dict[str, Any]],
+        logger: Optional[logging.Logger] = None,
+        minio_client: Optional[Any] = None,
+    ):
+        self.logger = logger or logging.getLogger(__name__)
+        self.minio_client = minio_client
+
+        if isinstance(cfg, str) and cfg.startswith("s3a://") and self.minio_client is None:
+            self.logger.error("❌ MinIO client must be provided for s3a:// paths.")
+            raise ValueError("MinIO client must be provided when loading configuration from an s3a:// path.")
+
+        try:
+            self.config: Dict[str, Any] = self._load_config(cfg)
+            self.logger.info("✅ Config loaded successfully")
+            self._validate_minimal_fields()
+        except Exception as e:
+            self.logger.error(f"❌ ConfigLoader initialization failed: {e}", exc_info=True)
+            raise
+
+    # ----------------------------------------------------------------------
+    # Internal helpers
+    # ----------------------------------------------------------------------
+    def _load_config(self, cfg: Union[str, Dict[str, Any]]) -> Dict[str, Any]:
+        """Load config from dict, local file, MinIO path, or JSON string."""
+        if isinstance(cfg, dict):
+            self.logger.info("📄 Using inline dict configuration")
+            return cfg
+
+        # --- Local file ---
+        if os.path.exists(cfg):
```

Comment:
Possible path traversal vulnerability - Users could potentially access files outside intended directories - we should add some path validation.